src/ngraph_deassign_clusters.cc

```diff
@@ -57,19 +57,58 @@ static void MaybeLogPlacement(const Graph* graph) {
   if (!config::IsLoggingPlacement()) return;
 
   std::map<int, std::set<const Node*>> final_cluster_map;
-
+  int number_of_nodes = 0, nodes_marked_for_clustering = 0,
+      nodes_assigned_cluster = 0;
   for (auto node : graph->nodes()) {
+    number_of_nodes++;
+    // Check marked for clustering
+    if (NodeIsMarkedForClustering(node)) {
+      nodes_marked_for_clustering++;
+    }
+
+    // Check Cluster Assignment
     int cluster_idx;
     if (!GetNodeCluster(node, &cluster_idx).ok()) {
       cluster_idx = -1;
+    } else {
+      nodes_assigned_cluster++;
     }
     final_cluster_map[cluster_idx].insert(node);
   }
 
+  int perc_marked_for_clustering_of_total =
+      (int)(((float)nodes_marked_for_clustering) * 100.0 / number_of_nodes);
```

Comment:
Do you need explicit type casting to float and then to int?
Wouldn't (int)/(int) * (int) give an (int) ?